test/gui/shared/steps/steps.py

```diff
@@ -535,3 +532,77 @@ def step(context, resource):
 )
 def step(context, resource, role):
     createPublicShareWithRole(context, resource, role)
+
+
+@When('the user logs out of the client-UI')
+def step(context):
+    toolbar = Toolbar()
+    toolbar.userLogout()
+
+
+def isUserSignedOut(context, username):
+    displayname = getDisplayname(username)
+    server = context.userData['localBackendUrl']
+    toolbar = Toolbar()
+    test.compare(
+        str(waitForObjectExists(toolbar.SIGNED_OUT_TEXT_BAR).text),
+        'Signed out from <a href="'
+        + server
+        + '">'
+        + server
+        + '</a> as <i>'
+        + displayname
+        + '</i>.',
+    )
+
+
+def isUserSignedIn(context, username):
+    displayname = getDisplayname(username)
+    server = context.userData['localBackendUrl']
+    toolbar = Toolbar()
+
+    test.compare(
+        str(waitForObjectExists(toolbar.SIGNED_OUT_TEXT_BAR).text),
+        'Connected '
+        + 'to <a href="'
+        + server
+        + '">'
+        + server
+        + '</a> as <i>'
+        + displayname
+        + '</i>.',
+    )
+
+
+@Then('user "|any|" should be signed out')
+def step(context, username):
+    isUserSignedOut(context, username)
+
+
+@Given('user "|any|" has logged out of the client-UI')
+def step(context, username):
+    waitFor(
+        lambda: isFolderSynced(context.userData['clientSyncPath']),
+        context.userData['clientSyncTimeout'] * 1000,
+    )
+    snooze(5)
```

Comment:
is there any specific reason for this snooze? Everything else looks good to me.

Reply:
It was added to see if folder was synced or not. Now there is no need for it so I have removed it.